codebase2/codebase-sqlite/U/Codebase/Sqlite/NamedRef.hs

```diff
@@ -44,3 +44,18 @@ toRowWithNamespace :: ToRow ref => NamedRef ref -> [SQLData]
 toRowWithNamespace nr = toRow nr <> [SQLText namespace]
   where
     namespace = Text.intercalate "." . reverse . NEL.tail . reversedSegments $ nr
+
+-- | The new 'scoped' name lookup format is different from the old version.
+--
+-- Specifically, the scoped format adds the 'lastNameSegment' as well as adding a trailing '.' to the db format
+-- of both the namespace and reversed_name.
+--
+-- Converts a NamedRef to SQLData of the form:
+-- [reversedName, namespace, lastNameSegment] <> ref fields...
+namedRefToScopedRow :: ToRow ref => NamedRef ref -> [SQLData]
```

Comment:
Prepares a named ref for the DB in the format the new lookups expect, including the new trailing `.`s